File: backend/app/schemas/product.py
"""商品相关模型"""
from pydantic import BaseModel


class NutritionItem(BaseModel):
    name: str
    value: str
    unit: str
    per: str = "100g"


class ProductOut(BaseModel):
    id: int
    barcode: str
    name: str
    brand: str = ""
    category: str = ""
    image_url: str = ""
    description: str = ""
    ingredients: list[str] = []
    tags: list[str] = []
    nutrition: list[NutritionItem] = []


class PricePoint(BaseModel):
    date: str
    price: float


class PriceHistoryOut(BaseModel):
    product_id: int
    barcode: str = ""
    points: list[PricePoint] = []

File: frontend/src/api/product.ts
import request from './request'

export interface NutritionItem {
  name: string
  value: string
  unit: string
  per: string
}

export interface Product {
  id: number
  barcode: string
  name: string
  brand: string
  category: string
  image_url: string
  description: string
  ingredients: string[]
  tags: string[]
  nutrition: NutritionItem[]
}

export interface PricePoint {
  date: string
  price: number
}

export interface PriceHistory {
  product_id: number
  barcode: string
  points: PricePoint[]
}

export const getProductApi = (barcode: string) =>
  request.get<Product>(`/products/${barcode}`).then((r) => r.data)

export const getPriceHistoryApi = (productId: number) =>
  request.get<PriceHistory>(`/products/${productId}/price-history`).then((r) => r.data)
